integration-libs/opf/base/core/services/opf-pickup-in-store-handler.service.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable, inject } from '@angular/core';
+import { ActiveCartFacade } from '@spartacus/cart/base/root';
+import { Product } from '@spartacus/core';
+import {
+  OpfQuickBuyDeliveryInfo,
+  OpfQuickBuyDeliveryType,
+} from '@spartacus/opf/base/root';
+import { IntendedPickupLocationFacade } from '@spartacus/pickup-in-store/root';
+import { CurrentProductService } from '@spartacus/storefront';
+import { Observable, of } from 'rxjs';
+import { map, switchMap, take } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class OpfPickupInStoreHandlerService {
+  protected currentProductService = inject(CurrentProductService);
```

Comment:
[MAJOR]

IMHO the `OpfPickupInStoreHandlerService` should not be tightly coupled to the PDP page and it should not depend on the `currentProductService`. We should be able to use it on any page, e.g. in PLP.

So I suggest to instead the method `getSingleProductDeliveryinfo` should take an argument with a `Product` object.

Reply:
[OUTDATED] We do not use such service anymore.